snapcraft/parts/lifecycle.py

```diff
@@ -525,6 +525,8 @@ def _run_in_provider(
         bind_ssh=parsed_args.bind_ssh,
         build_on=project.get_build_on(),
         build_for=project.get_build_for(),
+        http_proxy=getattr(parsed_args, "http_proxy", None),
+        https_proxy=getattr(parsed_args, "https_proxy", None),
```

Comment:
Can `parsed_args` reach this point with these attributes undefined?

Reply:
No, they should always exist in `parsed_args`.  If the user didn't pass them as command line arguments nor their environmental variables, then `argparse` instantiates them with a value of `None`.

Even if they didn't happen to exist in `parsed_args`, `getattr` still returns the default `None`.

Reply:
If the attribute is always there, getting it using `parsed_args.http_proxy` could be more natural and easier to read (like `parsed_args.bind_ssh` a few lines above).

Reply:
Makes sense.  I switched to your suggestion.